src/components/switch/bl-switch.css

```diff
@@ -0,0 +1,72 @@
+:host {
+  display: inline-block;
+  cursor: pointer;
+}
+
+span {
+  display: inline-block;
+}
+
+.switch {
+  --track-height: var(--bl-size-xl, 24px);
+  --track-width: var(--bl-size-4xl, 48px);
+  --thumb-offset: var(--bl-size-4xs, 2px);
+  --thumb-height: calc(var(--track-height) - calc(2 * var(--thumb-offset)));
+  --thumb-width: var(--thumb-height);
+  --outline-width: var(--bl-size-4xs, 2px);
+  --outline-offset: 1px;
+
+  /* TODO: use predefined animation duration once it is ready */
+  --animation-duration: var(--bl-switch-animation-duration, 300ms);
+
+  background-color: var(--bl-color-content-passive);
+  border-radius: var(--bl-border-radius-pill);
+  height: var(--track-height);
+  transition-property: background-color;
+  transition-duration: var(--animation-duration);
+  width: var(--track-width);
+}
+
+.switch > span {
+  background-color: white;
+  border-radius: var(--bl-border-radius-circle);
+  height: var(--thumb-height);
+  left: var(--thumb-offset);
+  position: relative;
+  top: var(--thumb-offset);
+  transition: transform;
+  transition-duration: var(--animation-duration);
+  width: var(--thumb-width);
+}
+
+:host([checked]) .switch {
+  background-color: var(--bl-switch-color, var(--bl-color-primary));
```

Comment:
@leventozen it's not a local variable. We intentionally wanted to give the option to use another color for switches other than just the primary color. As documented in #266

Reply:
Oh okay, mb 👍 However, if we define it at the upper level, it would be more clear. What do you think?

Reply:
Since it's used only here in whole CSS file, I think it's ok to *not* define a local variable for it. We document it in the code and storybook anyway.

Reply:
I agree with you about it, but I feel like we break our variable definition standards here 🤔 I tend to think like that when I see it first.

Reply:
I agree with Levent here, even though it is not used at multiple places currently, I think we should still declare a local variable in the upper level always for readability and maintainability purpose, because:

(1) Let's say we extend this component in the future, and need to use the same value again then we will have to anyway do it. So it's better to have a habit of declaring it always.
(2) Devs will know which variables are used in whole CSS by just looking at the upper level.

Reply:
Yeah I agree. Using a local variable that reads `--bl-switch-color` (with fallback to primary color) would be helpful.